File: Tyche.Android/pool/src/main/java/com/felipearpa/pool/application/GetPoolGamblerScoresByPoolUseCase.kt
package com.felipearpa.pool.application

import com.felipearpa.pool.domain.PoolGamblerScoreRepository
import javax.inject.Inject

class GetPoolGamblerScoresByPoolUseCase
@Inject constructor(private val poolGamblerScoreRepository: PoolGamblerScoreRepository) {

    suspend fun execute(poolId: String, next: String? = null, searchText: String? = null) =
        poolGamblerScoreRepository.getPoolGamblerScoresByPool(
            poolId = poolId,
            next = next,
            searchText = searchText
        )
}

File: Tyche.iOS/Pool/Sources/Pool/Application/GetPoolGamblerScoresByPoolUseCase.swift
import Core

public class GetPoolGamblerScoresByPoolUseCase {
    private let poolGamblerScoreRepository: PoolGamblerScoreRepository
    
    public init(poolGamblerScoreRepository: PoolGamblerScoreRepository) {
        self.poolGamblerScoreRepository = poolGamblerScoreRepository
    }
    
    func execute(
        poolId: String,
        next: String? = nil,
        searchText: String? = nil
    ) async -> Result<CursorPage<PoolGamblerScore>, Error> {
        return await poolGamblerScoreRepository.getPoolGamblerScoresByPool(
            poolId: poolId,
            next: next,
            searchText: searchText
        )
    }
}
